src/cli-util.js

```diff
@@ -404,6 +410,77 @@ function createOptionUsageType(option) {
   }
 }
 
+function createDetailedUsage(optionName) {
+  const options = getOptionsWithOpposites(constant.detailedOptions);
+  const optionNames = options.map(option => option.name);
+
+  let optionIndex = optionNames.indexOf(optionName);
+
+  if (optionIndex === -1) {
+    const suggestionOptionIndex = optionNames.findIndex(
+      currentOptionName => leven(currentOptionName, optionName) < 3
+    );
+
+    if (suggestionOptionIndex !== -1) {
+      const suggestionOptionName = options[suggestionOptionIndex].name;
+      console.warn(
+        `Unexpected option name "${optionName}", did you mean "${suggestionOptionName}"?\n`
+      );
+      optionName = suggestionOptionName;
+      optionIndex = suggestionOptionIndex;
+    } else {
+      throw new Error(`Unexpected option name "${optionName}"`);
+    }
+  }
+
+  const option = options[optionIndex];
```

Comment:
Perhaps extract the above into a function so that we can do `const option = fooGetOption(optionName);` here?